lib/zclient.c

```diff
@@ -2480,24 +2491,28 @@ struct interface *zebra_interface_link_params_read(struct stream *s,
 	if (link_params_set_value(s, ifp) != 0)
 		goto stream_failure;
 
-	if (changed == NULL)
-		return ifp;
-
-	iflp = if_link_params_get(ifp);
+	if (changed != NULL) {
+		iflp = if_link_params_get(ifp);
 
-	if (iflp_prev_set && iflp) {
-		if (memcmp(&iflp_prev, iflp, sizeof(iflp_prev)))
-			*changed = true;
-		else
+		if (iflp_prev_set && iflp) {
+			if (if_link_params_cmp(&iflp_prev, iflp))
+				*changed = false;
+			else
+				*changed = true;
+		} else if (!iflp_prev_set && !iflp)
```

Comment:
I would prefer to keep memcmp() call and add one more for the extended administrative group if set rather than calling another function which compare field by field for performance.

Reply:
code changed